shellcheck-all.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+TMP_DIR=smw-build-tmp
+SCRIPTS_FILE="$TMP_DIR"/bash-scripts.txt
+ALL_PASSED_SC=true
+
+mkdir -p "$TMP_DIR"
+
+find . -not -path './buildroot/*' -name '*.sh' -type f > "$SCRIPTS_FILE"
+
+while read -r s; do
```

Comment:
It would be better to use find directly in the loop or to use globbing to get all sh files

Reply:
If you use find and shellcheck individually, then the script will fail and exit on the first file with shell errors/warnings. So a person modifying or adding a whole bunch of (failing) scripts that doesn't have shellcheck installed will have to fix it a couple of times before having all issues in their script solved. That is because each time they only know that only one additional file failed. Here if we check all and then fail then they will have a list of files that failed, so it saves them some time.

Reply:
Actually we don't need a loop here. shellcheck can have multiple arguments, e.g.

`shellcheck $(git ls-files '*.sh')

Alternatively you can also add flags to ls-files or just use find, ignoring output and buildroot.